repo/encryption/encryption_test.go

```diff
@@ -73,14 +75,119 @@ func TestRoundTrip(t *testing.T) {
 			}
 		})
 	}
 }
+
+func TestCiphertextSamples(t *testing.T) {
+	cases := []struct {
+		masterKey []byte
+		contentID []byte
+		payload   []byte
+		samples   map[string]string
+	}{
+		{
+			masterKey: []byte("01234567890123456789012345678901"), // 32 bytes
+			contentID: []byte("aabbccddeeffgghhiijjkkllmmnnoopp"), // 32 bytes
+			payload:   []byte("foo"),
+
+			// samples of base16-encoded ciphertexts of payload encrypted with masterKey & contentID
+			samples: map[string]string{
+				"NONE": hex.EncodeToString([]byte("foo")),
+
+				"AES256-GCM-HMAC-SHA256":        "785c71de7c8ae8a5c0b5e2ad03f0be21620329",
+				"CHACHA20-POLY1305-HMAC-SHA256": "c93d644c5de803f017cad8ca331b7331e4cf55",
+
+				// deprecated
+				"AES-128-CTR":  "54cd8d",
+				"AES-192-CTR":  "2d084b",
+				"AES-256-CTR":  "8a580a",
+				"SALSA20":      "bf5ec3",
+				"SALSA20-HMAC": "8bf37fd9ec69843c3c2ac2a2cfdd59f36077206a15289efde640d0e677d03e6ac8f8ec",
+			},
+		},
+		{
+			masterKey: []byte("01234567890123456789012345678901"), // 32 bytes
+			contentID: []byte("00000000000000000000000000000000"), // 32 bytes
+			payload:   []byte("quick brown fox jumps over the lazy dog"),
+
+			// samples of base16-encoded ciphertexts of payload encrypted with masterKey & contentID
+			samples: map[string]string{
+				"NONE": hex.EncodeToString([]byte("quick brown fox jumps over the lazy dog")),
+
+				"AES256-GCM-HMAC-SHA256":        "e485b1f970e5d31f74b81c5b6336c3c5ef0de8f507943ce402b8ad3f282b8fd2e0b2554b13d0274ae088e119e2823f435bff9723b8201d",
+				"CHACHA20-POLY1305-HMAC-SHA256": "3e539c14afbcb990a546404bd0f0cb4d92c7d56593e04338dbb035aa38a75df37fcc42ebbe348ef13a1a40afcb55b1e2e3834b529388c4",
+
+				// deprecated
+				"AES-128-CTR":  "974c5c1782076e3de7255deabe8706a509b5772a8b7a8e7f83d01de7098c945934417071ec5351",
+				"AES-192-CTR":  "1200e755ec14125e87136b5281957895eeb429be673b2241da261f949283aea59fd2fa64387764",
+				"AES-256-CTR":  "39f13367828efb5fb22b97865ca0dbaad352d0c1a3083ff056bc771b812239445ed8af022f3760",
+				"SALSA20":      "65ce12b14739aecbf9e6a9b9b9c4a72ffa8886fe0b071c0abdfb3d3e5c336b90f9af411ba69faf",
+				"SALSA20-HMAC": "a1dc47f250def4d97a422d505fb5e9a9a13699762cb32cfe7705982fa68ce71f54544ab932a1045fb0601087159954d563f0de0aaa15690d93ea63748bf91889e577daeeed5cf8",
+			},
+		}}
+
+	for _, tc := range cases {
+		verifyCiphertextSamples(t, tc.masterKey, tc.contentID, tc.payload, tc.samples)
+	}
+}
+
+func verifyCiphertextSamples(t *testing.T, masterKey, contentID, payload []byte, samples map[string]string) {
+	for _, encryptionAlgo := range encryption.SupportedAlgorithms(true) {
+		enc, err := encryption.CreateEncryptor(parameters{encryptionAlgo, masterKey})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		ct := samples[encryptionAlgo]
+		if ct == "" {
+			v, err := enc.Encrypt(payload, contentID)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !t.Failed() {
+				t.Errorf("missing ciphertext sample")
+			}
+
+			t.Logf("%q: %q,", encryptionAlgo, hex.EncodeToString(v))
```

Comment:
Since going down this block always fails the test, might as well collapse it to be explicit:

```suggestion
			t.Errorf("missing ciphertext sample for %q: %q,", encryptionAlgo, hex.EncodeToString(v))
```

Reply:
done